server/src/uuid/model/entity/mod.rs

```diff
@@ -405,6 +407,128 @@ impl Entity {
     }
 }
 
+impl Entity {
+    pub async fn add_revision<'a, E>(
+        payload: EntityAddRevisionPayload,
+        executor: E,
+    ) -> Result<Uuid, EntityAddRevisionError>
+    where
+        E: Executor<'a>,
+    {
+        let mut transaction = executor.begin().await?;
+
+        if let Err(..) = Entity::fetch_via_transaction(payload.entity_id, &mut transaction).await {
+            return Err(EntityAddRevisionError::EntityNotFound);
+        }
```

Comment:
That looks dangerous. The `UuidError` returned via `fetch_via_transaction` covers a couple of cases that don't necessarily mean that the entity does not exists, see https://github.com/serlo/serlo.org-database-layer/blob/main/server/src/uuid/model/uuid.rs#L42.

We should be specific here and correctly pass on other errors like `DatabaseError` so that the API does not wrongly tell the user that the entity does not exist.

Reply:
Fair enough.
I've mapped to two possible errors in the 306f7d9, altough `DatabaseError` is maybe to broad (that is, by the way, the error that comes in test, namely 'Cannot add or update a child row: a foreign key constraint fail').
I'm not sure anymore if we need an specific `EntityAddRevisionError::EntityNotFound`, or if we just make a `Entity::fetch_via_transaction` and throw the error up.

Reply:
It should be enough to handle `UuidError::NotFound` since `Entity::fetch_via_transaction` should already catch the relevant database errors :)

Reply:
alright, done in 463ad8a.
Thanks for your feedback